torch_xla/core/xrt_run_server.py

```diff
@@ -6,40 +6,74 @@
 import sys
 from pathlib import Path
 
-if __name__ == '__main__':
-  parser = argparse.ArgumentParser()
-  parser.add_argument(
-      '--port', type=str, help='Port that XRT local service will be using.')
-  parser.add_argument(
-      '--env',
-      action='append',
-      type=str,
-      help='List of environment variables to distribute.')
-  FLAGS = parser.parse_args()
+XRT_RUN_SERVER_PROCESS = 'torch_xla.core._xrt_run_server'
+XRT_SERVER_REGEX = '^python -m {} [0-9]+$'.format(XRT_RUN_SERVER_PROCESS)
+
+
+def server_is_alive():
+  return len(
+      subprocess.Popen(['pgrep', '-f', XRT_SERVER_REGEX],
+                       stdout=subprocess.PIPE).stdout.readline()) != 0
+
+
+def kill_service():
+  subprocess.call(['pkill', '-f', XRT_SERVER_REGEX])
+  # Wait unitl existing server process get killed.
+  found_server_process = False
+  while server_is_alive():
+    found_server_process = True
+    time.sleep(1)
+  # Server process might still hold the lock to the tpu device after turing into a zombie
+  # process with name <defunct>. Sleep a bit longer to make sure it exit completely.
+  if found_server_process:
+    time.sleep(5)
+
 
-  my_env = os.environ.copy()
+def run_service(port, flag_env):
+  if server_is_alive():
+    print('Server is already running, use --restart to restart the server.')
+    return
+
+  local_env = os.environ.copy()
   # Enable the basic logging by defualt
-  my_env['TF_CPP_MIN_LOG_LEVEL'] = '0'
-  my_env[
+  local_env['TF_CPP_MIN_LOG_LEVEL'] = '0'
+  local_env[
       'TF_CPP_VMODULE'] = 'tpu_configuration_ops=1,tpu_execute_op=1,tpu_compile_op=1,tpu_compile_op_impl=1,tpu_compile_op_common=1,tpu_compile_ops=1,master=1,computation_client=5'
 
-  env_vars = list(FLAGS.env) if FLAGS.env else []
+  env_vars = list(flag_env) if flag_env else []
   for env_var in env_vars:
     if re.match(r'\w*=\w*', env_var) is None:
       raise ValueError(('Environment variable to distribute ({}) should follow '
                         'the form: X=Y').format(env_var))
     (env, var) = env_var.split('=', 1)
-    my_env[env] = var
+    local_env[env] = var
 
   Path("/tmp/xrt_server_log").mkdir(parents=True, exist_ok=True)
   time_str = time.strftime("%Y%m%d-%H%M%S")
-  stderr_file = open("/tmp/xrt_server_log/server_err_{}.log".format(time_str),
-                     "w")
-  stdout_file = open("/tmp/xrt_server_log/server_out_{}.log".format(time_str),
-                     "w")
-  subprocess.Popen(
-      ["python", "-m", "torch_xla.core._xrt_run_server", FLAGS.port],
-      stdout=stdout_file,
-      stderr=stderr_file,
-      env=my_env,
-      start_new_session=True)
+  log_file = open("/tmp/xrt_server_log/server_{}.log".format(time_str), "w")
+  subprocess.Popen(["python", "-m", XRT_RUN_SERVER_PROCESS,
```

Comment:
On a second thought if the VM comes with `python -> python2` (which for example is the case for our 1VM image), the below would probably not work, fyi. Maybe we should specify `python3` explicitly, though it is hacky. wdyt?

Reply:
good point, I will use python3 instead